src/relayclient/GSNConfigurator.ts

```diff
@@ -63,7 +75,9 @@ export async function resolveConfigurationGSN (provider: Web3Provider, partialCo
     throw new Error('Cannot resolve GSN deployment without paymaster address')
   }
 
-  const contractInteractor = new ContractInteractor(provider, defaultGsnConfig)
+  const tmpConfig = Object.assign({}, partialConfig, defaultGsnConfig)
+  const logger = createLogger(tmpConfig.logLevel, tmpConfig.customerToken, tmpConfig.hostOverride, tmpConfig.userIdOverride)
```

Comment:
I thought the whole idea was to make the logger a "dependency", and have defaultLogger as loglevel logger object.
why is it any different?

Reply:
OG-283 addresses the issue. The core problem here is how `resolveConfigurationGSN` does not belong to any object, so has no access to already created subdependencies.